conjure-go-client/httpclient/client_params.go

```diff
@@ -122,15 +122,27 @@ func WithMiddleware(h Middleware) ClientOrHTTPClientParam {
 	})
 }
 
+// WithInnerMiddleware is like WithMiddleware, but adds the handler to the
+// beginning of the middleware chain. This function will see the request last
+// (after all already-configured middleware) and see the response first.
+// This is useful for middleware that wants to mutate the request, including
+// overwriting actions from previous middleware.
+func WithInnerMiddleware(h Middleware) ClientOrHTTPClientParam {
```

Comment:
I also considered calling this something like `WithRequestMiddleware` or `WithRequestModifier` and making the signature `func(req *http.Request) error` instead of the full middleware that gets the ResponseWriter and RoundTripper. It would be less flexible (for better or worse) than this which allows getting the `*http.Response` before anyone else, but could also let us run these outside the middleware stack (to shrink the currently-huge stacktrace)